### PR TITLE
test/e2e: use vSphere projects from Boskos

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
For context see: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3034

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -42,6 +85,9 @@ export VSPHERE_PASSWORD="${GOVC_PASSWORD}"
 export VSPHERE_SSH_AUTHORIZED_KEY="${VM_SSH_PUB_KEY}"
 export VSPHERE_SSH_PRIVATE_KEY="/root/ssh/.private-key/private-key"
 
+# FIXME(sbueringer) before merge this should be moved to the ProwJob config in test-infra (so we can move jobs incrementally to the new environment / boskos)
+export BOSKOS_HOST=http://192.168.6.138:32222
```
Comment (sbueringer) [inline hack/e2e.sh:89]: /hold

must be removed before merge

**Comment by sbueringer** [comment]
cc @chrischdi @fabriziopandini 
(this is atm a POC to see if everything works in CPI as well)

**Comment by sbueringer** [comment]
/test pull-cloud-provider-vsphere-e2e-test
/test pull-cloud-provider-vsphere-e2e-test-on-latest-k8s-version

(forgot to change the network)

**Comment by sbueringer** [comment]
/assign @fabriziopandini @chrischdi @XudongLiuHarold 

Ready for review :0

File: hack/e2e.sh [hunk 2]
```diff
@@ -21,16 +21,59 @@ set -o pipefail # any non-zero exit code in a piped command causes the pipeline
 export PATH=${PWD}/hack/tools/bin:${PATH}
 REPO_ROOT=$(git rev-parse --show-toplevel)
 
+# In CI, ARTIFACTS is set to a different directory. This stores the value of
+# ARTIFACTS i1n ORIGINAL_ARTIFACTS and replaces ARTIFACTS by a temporary directory
```
Comment (lubronzhan) [inline hack/e2e.sh:25]:
```suggestion
# ARTIFACTS in ORIGINAL_ARTIFACTS and replaces ARTIFACTS by a temporary directory
```

**Comment by sbueringer** [comment]
/hold
only until https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3034 is merged

/assign @lubronzhan @XudongLiuHarold 
Please take a look already, so we can merge once the CAPV PR is merged.

Please let me know if you need more information than is provided on the linked issues

/assign @fabriziopandini @chrischdi

**Review by XudongLiuHarold — COMMENTED** [review]
/lgtm

generally looks great, thanks @sbueringer !

We might want to change this docker image: `gcr.io/k8s-staging-capi-vsphere/extra/boskosctl:latest` to cloud-provider-vsphere registry to avoid overlap, but I can do that in a follow up PR once I pushed this image.

**Comment by sbueringer** [comment]
> We might want to change this docker image: `gcr.io/k8s-staging-capi-vsphere/extra/boskosctl:latest` to cloud-provider-vsphere registry to avoid more coupling with capv, but I can do that in a follow-up PR once I pushed this image.

I would prefer to use the same image as we use in CAPV. That allows us - as CAPV maintainers - to easily maintain the vSphere environment for CAPV, image-builder and cloud-provider-vsphere.

**Comment by sbueringer** [comment]
@XudongLiuHarold @lubronzhan 
Can we get an approval for this PR? :) Thank you!

Please also put this on the list of PRs that have to be backported to all branches for which you still want to run CI

(like: https://github.com/kubernetes/cloud-provider-vsphere/pull/1063)

**Review by XudongLiuHarold — COMMENTED** [review]
/lgtm

/approve

Thanks @sbueringer !

**Pull request merged**